client/src/main/java/io/streamnative/oxia/client/lock/LightWeightLock.java

```diff
@@ -0,0 +1,478 @@
+                        },
+                        callbackService);
+            } else if (status == RELEASING) {
+                return runAsync(
+                        () -> {
+                            // switch to callback thread here
+                            throw wrap(new IllegalLockStatusException(INIT, RELEASING));
+                        },
+                        callbackService);
+            } else if (status == RELEASED) {
+                STATE_UPDATER.set(this, INIT);
+            } else {
+                return runAsync(
+                        () -> {
+                            // switch to callback thread here
+                            throw wrap(new LockException.UnknownLockStatusException(status));
+                        },
+                        callbackService);
+            }
+        }
+    }
+
+    private CompletableFuture<Void> tryLock1(long version) {
+        final PutOption versionOption =
+                version == Version.KeyNotExists
+                        ? PutOption.IfRecordDoesNotExist
+                        : PutOption.IfVersionIdEquals(versionId);
+        return client
+                .put(key, DEFAULT_VALUE, Set.of(PutOption.AsEphemeralRecord, versionOption))
+                .thenAccept(
+                        result -> {
+                            LightWeightLock.this.versionId = result.version().versionId();
+                            LightWeightLock.this.sessionId = result.version().sessionId();
+                            if (log.isDebugEnabled()) {
+                                log.debug(
+                                        "Acquired Lock. key={} session={} client_id={}",
+                                        key,
+                                        sessionId,
+                                        clientIdentifier);
+                            }
+                            STATE_UPDATER.set(this, ACQUIRED);
+                        })
+                .exceptionally(
+                        ex -> {
+                            final Throwable rc = unwrap(ex);
+                            final LockException lockE;
+                            if (rc instanceof UnexpectedVersionIdException
+                                    || rc instanceof KeyAlreadyExistsException) {
+                                lockE = new LockBusyException();
+                            } else {
+                                lockE = LockException.wrap(ex);
+                            }
+                            // ensure status rollback after exceptional future complete
+                            STATE_UPDATER.set(this, RELEASED);
+                            throw wrap(lockE);
+                        });
+    }
+
+    @Override
+    public CompletableFuture<Void> tryLock(
+            long time, TimeUnit unit, ExecutorService callbackService) {
+        return lock(callbackService)
+                .orTimeout(time, unit)
```

Comment:
When lock with retry, and put the retry task into taskService queue and wait for the next time schedule, but the tryLock timeout condition hit. It will fail this trylock operation, but the next retry won't cancel and it may succeed.

Reply:
Yes, that's a good point. let me fix it then. :)

Reply:
I deleted the `tryLock` with time now. and will support safe cancel future in the next PR.